File: flutter_app/lib/models/adherent.dart
class Adherent {
  final int idAdherent;
  final String nom;
  final String prenom;
  final String? dateNaissance;
  final String? sexe;
  final double? taille;
  final double? poids;
  final String? telephone;
  final String? objectif;
  final int idCoach;
  final String createdAt;

  final int? dernierScore;
  final String? derniereSession;

  const Adherent({
    required this.idAdherent,
    required this.nom,
    required this.prenom,
    this.dateNaissance,
    this.sexe,
    this.taille,
    this.poids,
    this.telephone,
    this.objectif,
    required this.idCoach,
    required this.createdAt,
    this.dernierScore,
    this.derniereSession,
  });


  factory Adherent.fromJson(Map<String, dynamic> json) {
    return Adherent(
      idAdherent: json['id_adherent'] ?? 0,

      nom: json['nom'] ?? '',

      prenom: json['prenom'] ?? '',

      dateNaissance: json['date_naissance'],

      sexe: json['sexe'],

      taille: (json['taille'] as num?)?.toDouble(),

      poids: (json['poids'] as num?)?.toDouble(),

      telephone: json['telephone'],

      objectif: json['objectif'],

      idCoach: json['id_coach'] ?? 0,

      createdAt: json['created_at'] ?? '',

      dernierScore: json['dernier_score'],

      derniereSession: json['derniere_session'],
    );
  }


  String get fullName => '$prenom $nom';
}

File: backend/app/database/models/adherent.py
from sqlalchemy import Column, Integer, String, Date, Float, DateTime, ForeignKey, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import enum

from app.database.database import Base


class SexeEnum(str, enum.Enum):
    M = "M"
    F = "F"


class Adherent(Base):
    __tablename__ = "adherents"

    id_adherent = Column(Integer, primary_key=True, autoincrement=True)
    nom = Column(String(100), nullable=False)
    prenom = Column(String(100), nullable=False)
    date_naissance = Column(Date, nullable=True)
    sexe = Column(Enum(SexeEnum), nullable=True)
    taille = Column(Float, nullable=True)  # en cm
    poids = Column(Float, nullable=True)  # en kg
    telephone = Column(String(20), nullable=True)
    objectif = Column(String(255), nullable=True)
    created_at = Column(DateTime, server_default=func.now())

    # FK
    id_coach = Column(Integer, ForeignKey("coachs.id_coach"), nullable=False)

    # Relations
    coach = relationship("Coach", back_populates="adherents")
    seances_ems = relationship("SeanceEms", back_populates="adherent")